packages/vow/src/vow.js

```diff
@@ -129,17 +136,29 @@ export const prepareVowKit = zone => {
       },
       watchNextStep: {
         onFulfilled(value) {
-          const { resolver } = this.facets;
+          const { resolver, watchNextStep } = this.facets;
           const { resolve } = getPromiseKitForResolution(resolver);
           if (resolve) {
             resolve(value);
           }
           this.state.stepStatus = 'fulfilled';
-          this.state.value = value;
+          if (zone.isStorable(value)) {
+            this.state.value = value;
+          } else {
+            watchNextStep.onRejected(
+              assert.error(`Vow fulfillment value is not storable: ${value}`),
```

Comment:
We need the following at the top of the file:
```js
const { details: X } = assert;
```

then:
```suggestion
              assert.error(X`Vow fulfillment value is not storable: ${value}`),
```